src/ut/cache_test.cpp

```diff
@@ -452,38 +535,84 @@ MATCHER_P(OperationHasResult, expected_rc, "")
   return (expected_rc == actual_rc);
 }
 
-TEST_F(CacheRequestTest, PutTransportEx)
+
+TEST_F(CacheRequestTest, PutOneTransportEx)
+{
+  TestTransaction *trx = make_trx();
+  Cache::PutRegData* put_reg_data = _cache.create_PutRegData("kermit", 1000);
+  put_reg_data->with_xml("<xml>");
+
+  apache::thrift::transport::TTransportException te;
+
+  {
+    testing::InSequence s;
+
+    EXPECT_CALL(_client, batch_mutate(_, _)).WillOnce(Throw(te)).RetiresOnSaturation();
+    EXPECT_CALL(_resolver, blacklist(_targets[0])).Times(1);
+    EXPECT_CALL(_client, batch_mutate(_, _)).WillOnce(Return());
+    EXPECT_CALL(_resolver, success(_targets[1])).Times(1);
+  }
+
+  EXPECT_CALL(*trx, on_success(_));
+  EXPECT_CALL(*_cm, inform_success(_));
+  execute_trx((CassandraStore::Operation*)put_reg_data, trx);
+}
+
+
+TEST_F(CacheRequestTest, PutTwoTransportEx)
 {
   TestTransaction *trx = make_trx();
   Cache::PutRegData* put_reg_data = _cache.create_PutRegData("kermit", 1000);
   put_reg_data->with_xml("<xml>");
 
   apache::thrift::transport::TTransportException te;
-  EXPECT_CALL(_cache, get_client()).Times(2).WillRepeatedly(Return(&_client));
-  EXPECT_CALL(_cache, release_client()).Times(2);
+  EXPECT_CALL(_resolver, blacklist(_targets[0])).Times(1);
+  EXPECT_CALL(_resolver, blacklist(_targets[1])).Times(1);
   EXPECT_CALL(_client, batch_mutate(_, _)).Times(2).WillRepeatedly(Throw(te));
 
   EXPECT_CALL(*trx, on_failure(OperationHasResult(CassandraStore::CONNECTION_ERROR)));
   EXPECT_CALL(*_cm, inform_failure(_));
   execute_trx((CassandraStore::Operation*)put_reg_data, trx);
 }
 
-TEST_F(CacheRequestTest, PutTransportGetClientEx)
+
+TEST_F(CacheRequestTest, PutTransportConnectEx)
```

Comment:
This test throws two exceptions in different places. The testcase name could do with clarifying.